pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/batch/standalone/SegmentGenerationJobRunner.java

```diff
@@ -194,6 +199,23 @@ public void run()
         filteredFiles.add(file);
       }
     }
+
+    // Sort files based on creation time.
+    Collections.sort(filteredFiles, new Comparator<String>() {
```

Comment:
Suggest not addressing this TODO in this PR. It can potentially make lots of calls (O(nlogn)) to the PinotFS and cause issues when it is remote. Recently we have encountered issues of hitting the request rate limit for S3.

Reply:
I had to sort files to get deterministic ordering for the test, and since there was a TODO I went ahead and made the change. I can reduce to O(N) by caching the timestamps where we already make the `.isDirectory()` call, lmk if that's an acceptable compromise.

Reply:
OK, since input files are sorted by name if `useGlobalDirectorySequenceId` isn't true, I'll rely on that to control ordering of segment generation in the test. I've removed the sort code, and put a comment in about sorting being problematic given the re-ordering that can happen, and load on external file systems.